src/measure_theory/measure/with_density_vector_measure.lean

```diff
@@ -142,4 +142,24 @@ lemma integrable.with_densityᵥ_eq_iff {f g : α → E}
   μ.with_densityᵥ f = μ.with_densityᵥ g ↔ f =ᵐ[μ] g :=
 ⟨λ hfg, hf.ae_eq_of_with_densityᵥ_eq hg hfg, λ h, with_densityᵥ_eq.congr_ae h⟩
 
+section signed_measure
+
+lemma with_densityᵥ_eq_with_density_pos_part_sub_with_density_neg_part
+  {f : α → ℝ} (hfi : integrable f μ) :
+  μ.with_densityᵥ f =
+  @to_signed_measure α _ (μ.with_density (λ x, ennreal.of_real $ f x))
+    (is_finite_measure_with_density_of_real hfi.2) -
+  @to_signed_measure α _ (μ.with_density (λ x, ennreal.of_real $ -f x))
+    (is_finite_measure_with_density_of_real hfi.neg.2) :=
+begin
+  ext i hi,
+  rw [with_densityᵥ_apply hfi hi, integral_eq_lintegral_pos_part_sub_lintegral_neg_part,
+      vector_measure.sub_apply, to_signed_measure_apply_measurable hi,
+      to_signed_measure_apply_measurable hi, with_density_apply _ hi, with_density_apply _ hi],
+  rw ← integrable_on_univ at hfi ⊢,
+  exact hfi.restrict measurable_set.univ
+end
```

Comment:
The two lines at the bottom of your proof can be replaced by `exact hfi.integrable_on`.
```suggestion
begin
  ext i hi,
  rw [with_densityᵥ_apply hfi hi,
    integral_eq_lintegral_pos_part_sub_lintegral_neg_part hfi.integrable_on,
    vector_measure.sub_apply, to_signed_measure_apply_measurable hi,
    to_signed_measure_apply_measurable hi, with_density_apply _ hi, with_density_apply _ hi],
end
```